### DISCUSSION TITLE
Asking for administrator's permission

### DISCUSSION THREAD

**Discussion by Ahelgesson** [MiczFlor/RPi-Jukebox-RFID#2370] [Q&A] [answered]
Hi. Quite new to all this so sorry if i misunderstood something basic. I have built a Phoniebox using v2 and am quite happy with it. I have added songs, rfid reader and its working better than intended. 

The roadblock is that I wanted to add some buttons to the box and had a problem with the config file. I have found the gpio_settings config file inside pi\RPi-Jukebox-RFID\settings but I don't seem to be able to change/save or do anything with it (getting error: need to ask for administrator's permission). 

If I try to give myself access using the "Give Security permission" function in network settings it says I don't have permission to do that. I have tried to change the file from two different computers at this time (windows and chromebook) and while both find the file without problem and don't seem to have access to edit it. 

Would appreciate any help

**Comment by AlvinSchiller** [comment]
Hi,
from your description it sound like you are using the desktop version of raspberry pi OS? Which version do you use exactly?
And which editor do you use to edit? 


Could you run this command in a terminal  `ls -l /home/pi/RPi-Jukebox-RFID/settings` and share the output?

**Reply by Ahelgesson** [reply]
Thanks for the input! 

Im using "pi os (legacy 32-bit) lite"
as for the editor i tried both wordpad and visual studios

And here is the output:

```
pi@julia:~ $ ls -l /home/pi/RPi-Jukebox-RFID/settings
total 160
-rwxrwxr-x 1 pi       www-data   46 May  9 08:55 Audio_Folders_Path
-rwxrwxrwx 1 pi       pi          2 May  9 08:57 Audio_iFace_Active
-rwxrwxr-x 1 pi       www-data    4 May  9 08:55 Audio_iFace_Name
-rwxrwxr-x 1 pi       www-data    2 May  9 08:55 Audio_Volume_Change_Step
-rw-r--r-- 1 pi       pi          5 May  9 08:57 Change_Volume_Idle
-rwxrwxrwx 1 pi       pi         16 May  9 08:57 CHAPTEREXTENSIONS
-rwxrwxrwx 1 pi       pi          4 May  9 08:57 CHAPTERMINDURATION
-rwxrwxr-x 1 pi       www-data  446 May  9 08:55 debugLogging.conf
-rwxrwxr-x 1 pi       www-data  446 May  9 08:46 debugLogging.conf.sample
-rwxrwxrwx 1 pi       www-data    8 May  9 08:55 edition
-rwxrwxrwx 1 pi       www-data  756 May 13 08:11 global.conf
-rwxrwxr-x 1 pi       www-data 2666 May  9 08:55 gpio_settings.ini
-rwxrwxr-x 1 pi       www-data    2 May  9 08:55 Idle_Time_Before_Shutdown
-rwxrwxrwx 1 pi       pi          6 May  9 08:57 Lang
-rwxrwxrwx 1 pi       www-data    9 May 10 19:52 Latest_Folder_Played
-rwxrwxrwx 1 pi       www-data    9 May 10 19:52 Latest_Playlist_Played
-rw-r--r-- 1 pi       pi         11 May 10 19:52 Latest_RFID
-rwxrwxrwx 1 pi       pi          4 May  9 08:57 MailWlanIpYN
-rwxrwxr-x 1 pi       www-data    4 May  9 08:55 Max_Volume_Limit
-rwxrwxrwx 1 pi       pi          2 May  9 08:57 Min_Volume_Limit
-rwxrwxr-x 1 pi       www-data  349 May  9 08:55 PhonieboxInstall.conf
-rwxrwxr-x 1 pi       www-data   36 May  9 08:55 Playlists_Folders_Path
-rwxrwxrwx 1 pi       pi         14 May  9 08:57 Poweroff_Command
-rwxrwxrwx 1 pi       pi          4 May  9 08:57 Rfidreader_Rc522_Readmode_UID
-rwxrwxr-x 1 www-data www-data 5503 May  9 09:07 rfid_trigger_play.conf
-rwxrwxr-x 1 pi       www-data 5504 May  9 08:46 rfid_trigger_play.conf.sample
-rwxrwxr-x 1 pi       www-data    6 May  9 09:07 Second_Swipe
-rwxrwxrwx 1 pi       pi          2 May  9 09:05 Second_Swipe_Pause
-rwxrwxrwx 1 pi       pi          3 May  9 08:57 Second_Swipe_Pause_Controls
-rwxrwxr-x 1 pi       www-data    3 May  9 08:55 ShowCover
-rwxrwxrwx 1 pi       pi          2 May  9 08:57 Startup_Volume
-rwxrwxrwx 1 pi       pi         14 May  9 09:06 Swipe_or_Place
-rwxrwxr-x 1 pi       www-data   26 May 13 08:11 version
-rwxrwxr-x 1 pi       www-data    6 May  9 08:46 version-number
-rwxrwxrwx 1 pi       pi          3 May  9 08:57 Volume_Boot
-rwxrwxrwx 1 pi       pi          4 May  9 08:57 Volume_Manager
-rwxrwxrwx 1 pi       pi          1 May  9 08:57 WlanIpMailAddr
-rwxrwxrwx 1 pi       pi          4 May  9 08:57 WlanIpReadYN

```

**Reply by AlvinSchiller** [reply, marked as answer]
> Im using "pi os (legacy 32-bit) lite"

Ok then i don't know what you mean with "If I try to give myself access using the "Give Security permission" function in network settings it says I don't have permission to do that." ^^

> as for the editor i tried both wordpad and visual studios

So you only open them on a remote host? How do you access the Pi (WinScp, Samba, ...)?

> And here is the output:

That looks good so far.

Can you try and edit the file from a terminal?
- run `nano /home/pi/RPi-Jukebox-RFID/settings/gpio_settings.ini`
- make sure the correct file is opened
- change something (e.g. add a comment `; test write`)
- save the file with `CTRL` + `o`, then `Enter` to confirm the file name
- close the file with `CTRL` + `x`

**Reply by Ahelgesson** [reply]
yeah, for some reason it never occurred for me that you could edit from powershell. I only tried from a remote host. Your suggestion using nano worked like a charm! Thanks for all the help !

**Reply by AlvinSchiller** [reply]
Nevertheless should it be possible. I use WinSCP on Windows and edit the files with Notepad++ and i didn't have a problem with saving so far. As long as the file permissions are correctly set (like in your case they are) this should work.
